rstream/client.py

```diff
@@ -172,9 +173,34 @@ async def start(self) -> None:
         self._conn = Connection(self.host, self.port, self._ssl_context)
         await self._conn.open()
         self.start_task("listener", self._listener())
+
         self.add_handler(schema.Heartbeat, self._on_heartbeat)
         self.add_handler(schema.Close, self._on_close)
 
+    async def run_queue_listener_task(self, subscriber_name: str, handler: HT[FT]):
+
+        if subscriber_name not in self._frames:
+            self.start_task(
+                "run_delivery_handlers" + subscriber_name,
+                self._run_delivery_handlers(subscriber_name, handler),
+            )
+
+    async def _run_delivery_handlers(self, subscriber_name: str, handler: HT[FT]):
+
+        while True:
+            frame_entry = None
+            try:
+                frame_entry = await self._frames[subscriber_name].get()
+                maybe_coro = handler(frame_entry)
+                if maybe_coro is not None:
+                    await maybe_coro
+            except Exception as e:
+                if frame_entry is not None:
+                    logger.exception("Error while handling %s frame ", str(frame_entry.__class__))
+                else:
+                    logger.exception("Error while handling a frame " + str(e))
+                    break
```

Comment:
I'm not sure about this `break`, we probably shouldn't stop handling frames if there's an exception in a user-provided callback function

Reply:
It seems like this is the info we have (from github action)

```
=============================== warnings summary ===============================
tests/test_encoding.py::test_encoding[frame26]
  /home/runner/.cache/pypoetry/virtualenvs/rstream-KivhoGsP-py3.9/lib/python3.9/site-packages/_pytest/unraisableexception.py:78: PytestUnraisableExceptionWarning: Exception ignored in: <coroutine object BaseClient._run_delivery_handlers at 0x7f9b74cf68c0>
  
  Traceback (most recent call last):
    File "/home/runner/.cache/pypoetry/virtualenvs/rstream-KivhoGsP-py3.9/lib/python3.9/site-packages/_pytest/fixtures.py", line 642, in _compute_fixture_value
      with suppress(KeyError):
  RuntimeError: coroutine ignored GeneratorExit
  
    warnings.warn(pytest.PytestUnraisableExceptionWarning(msg))
```

Reply:
It seems like something is happening when we are terminating the task causing that issue.
Originally in my laptop the test encoding frame24 was pending for this reason:

```
WARNING  rstream.client:client.py:202 Error while handling a frame Event loop is closed
ERROR    asyncio:base_events.py:1771 Task was destroyed but it is pending!
task: <Task cancelling name='Task-726' coro=<BaseClient._run_delivery_handlers() running at /Users/dpalaia/projects/rstream/rstream/client.py:None> wait_for=<Future cancelled> cb=[BaseClient.start_task.<locals>.on_task_done() at /Users/dpalaia/projects/rstream/rstream/client.py:104]>
WARNING  rstream.client:client.py:191 im here
WARNING  rstream.client:client.py:202 Error while handling a frame <Queue at 0x110fbd150 maxsize=0 _getters[1] tasks=1> is bound to a different event loop
WARNING  rstream.client:client.py:191 im here
WARNING  rstream.client:client.py:202 Error while handling a frame <Queue at 0x110fbd150 maxsize=0 _getters[1] tasks=1> is bound to a different event loop
```

asyncio.Queue shouldn't raise exceptions when get is blocking. So I tried to put outside the except block.
In this case the test is not pending anymore but I still had this issue:

```
Traceback (most recent call last):
    File "/Users/dpalaia/projects/rstream/rstream/client.py", line 191, in _run_delivery_handlers
      frame_entry = await self._frames[subscriber_name].get()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/usr/local/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/queues.py", line 160, in get
      getter.cancel()  # Just in case getter is not done yet.
      ^^^^^^^^^^^^^^^
    File "/usr/local/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 761, in call_soon
      self._check_closed()
    File "/usr/local/Cellar/python@3.11/3.11.5/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 519, in _check_closed
      raise RuntimeError('Event loop is closed')
  RuntimeError: Event loop is closed
```

Maybe the best solution would be to rewrite the code in this way:
Removing the while true and terminating the task when we close() the client (in close() we also set self._is_not_closed to False.

```
        while self._is_not_closed:
            frame_entry = await self._frames[subscriber_name].get()
            try:
                maybe_coro = handler(frame_entry)
                if maybe_coro is not None:
                    await maybe_coro
            except Exception as e:
                logger.warning("Error while handling %s frame ", frame_entry.__class__)
```